components/Wizard/ImageStep.js

```diff
@@ -292,20 +295,141 @@ class ImageStep extends React.PureComponent {
+            min={minImageSize}
+            max={maxImageSize}
+            value={imageSize || ""}
+            validated={imageSizeValidated}
+            onChange={this.handleImageSizeInput}
+            isRequired
+          />
+          <InputGroupText>GB</InputGroupText>
+        </InputGroup>
+      </FormGroup>
+    );
+
+    const ostreeFields = (
+      <>
+        <FormGroup
+          label={<FormattedMessage defaultMessage="Parent commit" />}
+          labelIcon={
+            <Popover
+              id="ostree-parent-popover"
+              bodyContent={
+                <FormattedMessage defaultMessage="Provide the ID of the latest commit in the updates repository for which this commit provides an update." />
+              }
+              aria-label={formatMessage(ariaLabels.ostreeParent)}
+            >
+              <Button variant="plain" aria-label={formatMessage(ariaLabels.ostreeParent)}>
+                <OutlinedQuestionCircleIcon className="cc-c-text__align-icon" id="popover-icon" />
+              </Button>
+            </Popover>
+          }
+          fieldId="ostree-parent-input"
+          hasNoPaddingTop
+        >
+          <TextInput
+            className="pf-c-form-control"
+            value={ostreeSettings.parent !== undefined ? ostreeSettings.parent : ""}
+            type="text"
+            id="ostree-parent-input"
+            onChange={setOstreeParent}
+          />
+        </FormGroup>
+        <FormGroup
+          label={<FormattedMessage defaultMessage="Ref" />}
+          labelIcon={
+            <Popover
+              id="ostree-ref-popover"
+              bodyContent={
+                <FormattedMessage defaultMessage="Provide the name of the branch for the content. If the ref does not already exist it will be created." />
+              }
+              aria-label={formatMessage(ariaLabels.ostreeRef)}
+            >
+              <Button variant="plain" aria-label={formatMessage(ariaLabels.ostreeRef)}>
+                <OutlinedQuestionCircleIcon className="cc-c-text__align-icon" id="popover-icon" />
+              </Button>
+            </Popover>
+          }
+          fieldId="ostree-ref-input"
+          helperText={formatMessage(messages.ostreeRefHelperText)}
+          helperTextInvalid={formatMessage(messages.ostreeRefHelperText)}
+          validated={ostreeRefValidated}
+          hasNoPaddingTop
+        >
+          <TextInput
+            className="pf-c-form-control"
+            value={ostreeSettings.ref !== undefined ? ostreeSettings.ref : ""}
+            type="text"
+            id="ostree-ref-input"
+            aria-describedby="ostree-ref-input-helper"
+            onChange={this.handleOSTreeRef}
+            validated={ostreeRefValidated}
           />
-        </div>
-      </div>
+          {ostreeRefValidated === "default" && imageType === "rhel-edge-commit" && (
+            <p className="pf-c-form__helper-text" id="ostree-ref-input-helper" aria-live="polite">
```

Comment:
These updates look great regarding how it behaves and looks! 

There's just one minor issue regarding the `id` value defined for this `<p>` element. I included a screencap of what gets rendered, where you can see that both the `<div>` and `<p>` have the same `id` value, which isn't valid html. 

I'm guessing that the `<FormGroup>` component automatically generates the `id` for the `<div>` based on the value provided for `fieldID` a few lines above, so the `<p>` element will need a new id. And you can still reference this text in `aria-describedby` because this attribute can list more than one `id`, separated by spaces. 

So you can fix this by making the following updates:

- Provide a different value for this `<p>` element, e.g. `id="ostree-ref-input-helper-default"`
- For the `<TextInput>` component, reference both elements with help text, e.g. `aria-describedby="ostree-ref-input-helper-default ostree-ref-input-helper"`. 

<img width="333" alt="image" src="https://user-images.githubusercontent.com/21063328/92338123-32f8d280-f07c-11ea-9796-8b62a2290b90.png">

Reply:
Thanks for pointing this out. Should be fixed now.